### ISSUE TITLE
OpenSSL initialization is very slow

### ISSUE THREAD

**Issue by nirvdrum** [jruby/jruby#1597]
I don't think OpenSSL initialization being slow is news to anyone, but it keeps coming up as I've been trying to speed up start-up time.  On my i7-2760 QM (2.4 GHz quad-core w/ hyperthreading), a simple `require 'openssl'` can take 500 - 650 ms.  On a 2.7 GHz quad-core Opteron (virtualized), it takes 1.8s on average (although I have seen as high as 3s).

I've been working with various gem authors to push the `require 'openssl'` line as late into the process as possible.  But obviously there's a bit of push-back.

In lieu of speeding up the initialization time, perhaps it should at least be deferred so that the simple require statement is fast, but actual initialization is done upon first actual call.

**Comment by kares** [comment]
**0.9.6** cut things half and will be part of **1.7.19** ... there's certainly more optimization work possible, but maybe those should get split into smaller pieces instead of one general issue report here ...

**Comment by nirvdrum** [comment]
I'm fine scoping it more, but to clarify this is mostly aimed at all the static initialization that goes on.

**Comment by kares** [comment]
ah OK, been actually thinking about that but got slightly de-motivated by looking at RubyGems/Bundler to see what's it really using ... there's stuff to figure out from the "happy-path" to know what to load in advance.

**Comment by enebo** [comment]
I am resolving this as some progress was made on speeding up the loading but it never closed because it was open-ended (e.g. how fast should it get before we can say it is fast enough to close this issue).
